eng/native/configurecompiler.cmake

```diff
@@ -85,6 +85,7 @@ if (MSVC)
   #shrink pdb size
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /PDBCOMPRESS")
 
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /INCREMENTAL:NO")
```

Comment:
Why do we need to disable incremental build here?

Reply:
We were seeing toolchain failures without this (in particular, we were seeing https://github.com/dotnet/runtime/discussions/64054 locally).

Reply:
We also already were disabling incremental linking in other places, but CMake was adding it back.

Reply:
This didn't work btw. I was hoping I could get some pointers on how to get this to work. Currently, the workaround is to comment the cmake test that checks for the existence of `SIZEOF_DATA_PTR` in `src/native/external/zlib-ng/CMakeLists.txt`. But I would very much prefer to avoid making any changes to the zlib-ng code and figure out what we're missing.

Reply:
Looking at the problem again, you need to remove the `/INCREMENTAL` from the `CMAKE_EXE_LINKER_FLAGS_INIT` property. That's where it's getting added I believe.

Reply:
Fixed this with a different workaround (with a comment).

I'll see if I can contribute a better fix to CMake itself (which would take us a long time to consume, but still general goodness).